### ISSUE TITLE
rhel8 - Spyder crashing (DISA STIGed)

### ISSUE THREAD

**Issue by JosephHoffman** [spyder-ide/spyder#17802]
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X] Tried basic troubleshooting (if a bug/error)
    * [X] Restarted Spyder
    * [X] Reset preferences with ``spyder --reset``
    * [X] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [X] Tried the other applicable steps from the Troubleshooting Guide
* [X] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
Cannot launch Spyder on any Hardened (STIG) Red Hat Enterprise Linux 8 system. Receive errors when launching Spyder, as it crashes immediately. Tested on an non-hardened RHEL8 Workstation without issue. It seems to be a DISA STIG causing the issue. I am look for guidance on what could potentially be blocking based on the error output below.

### What steps reproduce the problem?

1. Hardened RHEL8 system (using Ansible-Lockdown project for RHEL8)
2. Installed the latest version of Anaconda, using the published .sh file follwing steps:
```
sudo su -
cd /root
mkdir -p /usr/local/src/tmpinstall
cp -vi Anaconda3-2021.11-Linux-x86_64.sh /usr/local/src/
TMPDIR=/usr/local/src/tmpinstall bash /usr/local/src/Anaconda3-2021.11-Linux-x86_64.sh -u -p /opt/anaconda3
vim .condarc
    Add:
        channel_alias: https://artifactory.URL/artifactory/api/conda
        channels:
          - conda-forge
          - conda-pytorch
          - anaconda-main
        default_channels:
          - https://artifactory.URL/artifactory/api/conda/conda-forge

/opt/anaconda3/bin/conda init bash
/opt/anaconda3/bin/conda update conda
/opt/anaconda3/bin/conda update --all
dnf install libXScrnSaver libXScrnSaver-devel
/opt/anaconda3/bin/spyder
```

### What is the expected output? What do you see instead?
Expected outcome is Spyder to launch. What we see instead:
```
[root@its-conda-test ~]# /opt/anaconda3/bin/spyder
Update LANGUAGE_CODES (inside config/base.py) if a new translation has been added to Spyder
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
link image0 hasn't been detected!
[120336:120336:0429/231430.943903:FATAL:credentials.cc(137)] Check failed: error == EPERM || error == EUSERS || error == EINVAL || error == ENOSYS. : No space left on device
#0 0x7fc4777680be <unknown>
#1 0x7fc47777a6c2 <unknown>
#2 0x7fc47777ba3c <unknown>
#3 0x7fc47813f845 <unknown>
#4 0x7fc47813fbe6 <unknown>
#5 0x7fc4774e41a5 <unknown>
#6 0x7fc47728e994 <unknown>
#7 0x7fc477292727 <unknown>
#8 0x7fc477124542 <unknown>
#9 0x7fc477124718 <unknown>
#10 0x7fc47711b681 <unknown>
#11 0x7fc47711b732 QtWebEngineCore::WebContentsAdapter::WebContentsAdapter()
#12 0x7fc47b9c8376 <unknown>
#13 0x7fc47b9c60c4 <unknown>
#14 0x7fc47b9c6252 QWebEnginePage::QWebEnginePage()
#15 0x7fc47b9f7689 sipQWebEnginePage::sipQWebEnginePage()
#16 0x7fc47b9f7812 init_type_QWebEnginePage
#17 0x7fc47cce98e4 <unknown>
#18 0x5611157089bb _PyObject_MakeTpCall
#19 0x56111578e8b4 _PyEval_EvalFrameDefault
#20 0x561115746b0d _PyEval_EvalCode
#21 0x561115747527 _PyFunction_Vectorcall
#22 0x56111573884e _PyObject_FastCallDictTstate
#23 0x5611157423d9 slot_tp_init
#24 0x5611157089bb _PyObject_MakeTpCall
#25 0x56111578f853 _PyEval_EvalFrameDefault
#26 0x561115746b0d _PyEval_EvalCode
#27 0x561115747527 _PyFunction_Vectorcall
#28 0x5611157387ca _PyObject_FastCallDictTstate
#29 0x5611157423d9 slot_tp_init
#30 0x5611157089bb _PyObject_MakeTpCall
#31 0x56111578e8b4 _PyEval_EvalFrameDefault
#32 0x561115746b0d _PyEval_EvalCode
#33 0x561115747527 _PyFunction_Vectorcall
#34 0x5611157387ca _PyObject_FastCallDictTstate
#35 0x5611157423d9 slot_tp_init
#36 0x5611157089bb _PyObject_MakeTpCall
#37 0x56111578e8b4 _PyEval_EvalFrameDefault
#38 0x561115746b0d _PyEval_EvalCode
#39 0x561115747527 _PyFunction_Vectorcall
#40 0x56111573884e _PyObject_FastCallDictTstate
#41 0x5611157423d9 slot_tp_init
#42 0x5611157089bb _PyObject_MakeTpCall
#43 0x56111578f853 _PyEval_EvalFrameDefault
#44 0x5611157739b3 method_vectorcall
#45 0x5611156be90b _PyEval_EvalFrameDefault.cold.2975
#46 0x561115746b0d _PyEval_EvalCode
#47 0x561115747527 _PyFunction_Vectorcall
#48 0x56111573884e _PyObject_FastCallDictTstate
#49 0x5611157423d9 slot_tp_init
#50 0x5611157089bb _PyObject_MakeTpCall
#51 0x56111578f853 _PyEval_EvalFrameDefault
#52 0x5611157739b3 method_vectorcall
#53 0x5611156bdfdc _PyEval_EvalFrameDefault.cold.2975
#54 0x5611157739b3 method_vectorcall
#55 0x5611156be90b _PyEval_EvalFrameDefault.cold.2975
#56 0x5611157739b3 method_vectorcall
#57 0x5611156bdfdc _PyEval_EvalFrameDefault.cold.2975
#58 0x561115747753 _PyFunction_Vectorcall
#59 0x5611156beebb _PyEval_EvalFrameDefault.cold.2975
#60 0x561115747753 _PyFunction_Vectorcall
#61 0x5611156bdfdc _PyEval_EvalFrameDefault.cold.2975

Aborted (core dumped)
```

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 5.1.5
* Python version: 3.9.7
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version: Red Hat Enterprise Linux (rhel) 8.5

### Dependencies

```
libXScrnSaver
libXScrnSaver-devel
```

**Comment by ccordoba12** [comment]
Hey @JosephHoffman, thanks for reporting. I have a couple of comments about what you reported:

1. I don't understand why you're trying to run Spyder as root. Why don't you run it as a regular user? I mean, this could be part of the problem.
2. Your problem seems similar to the one reported [here](https://github.com/spyder-ide/spyder/pull/16518#issue-1012685874). To avoid it, we added a CLI option called `--no-web-widgets` that's going to be available in our next version (5.3.1), to be released in a couple of weeks. So you should wait until it's available and try again.

**Comment by JosephHoffman** [comment]
Hello @ccordoba12,

1. I got the same exact results as multiple user accounts. Root was just the elevation I was when I grabbed the snippet. Account or privledge did not affect the results, typically we only run Spyder as the user.
2. When will 5.3.1 release? I have customers who need this functionality for projects they are working on, as I been struggling to find a resolution. This works fine on non-hardened (DISA STIG) RHEL8 workstations, and it works fine on RHEL7 Workstations regardless of hardened or non-hardened. I am just trying to determine if something is blocking Spyder, or perhaps permissions to a resource.
